life_expectancy/cleaning.py

```diff
@@ -0,0 +1,101 @@
+""" The code in this file loads, cleans and saves a dataset """
+from pathlib import Path
+import argparse
+import pandas as pd
+
+#--------------------------------------------------------------------
+# VARIABLES
+#--------------------------------------------------------------------
+
+CURRENT_DIR = str(Path(__file__).parent)
+PATH_RAW_DATASET = f"{CURRENT_DIR}/data/eu_life_expectancy_raw.tsv"
+PATH_CLEAN_DATASET = f"{CURRENT_DIR}/data/pt_life_expectancy.csv"
+new_col_names = ['unit', 'sex', 'age', 'region']
+
+#--------------------------------------------------------------------
+# FUNÇÔES
+#--------------------------------------------------------------------
+
+def load_data(path):
+    """ Loads the tsv dataset"""
+    dataset = pd.read_csv(path, sep='\t')
+
+    return dataset
+
+def split_first_four_columns(dataset):
+    """ Slipts the frist column of the dataset into four different columns """
+    old_col = 'unit,sex,age,geo\\time'
+
+    dataset[new_col_names] = dataset[old_col].str.split(',', expand=True)
+    dataset = dataset.drop(columns=[old_col])
+
+    return dataset
+
+def get_date_columns(dataset):
+    """ Gets all the columns that represent a date """
+    date_columns = dataset.drop(columns = new_col_names).columns
+
+    return date_columns
+
+def unpivot_date_columns(dataset):
+    """ Unpivots the date columns, turning them into data """
+    date_columns = get_date_columns(dataset)
+
+    unpivoted_df = pd.melt(dataset, value_vars = date_columns,\
+                                    id_vars = new_col_names,\
+                                    var_name = 'year',\
+                                    value_name='value')
+
+    return unpivoted_df
```

Comment:
You can avoid this variable with no scope by doing something like below. You don't need to name it :)

```suggestion
    return pd.melt(dataset, value_vars = date_columns,\
                                    id_vars = new_col_names,\
                                    var_name = 'year',\
                                    value_name='value')
```